src/main/java/org/vivoweb/harvester/util/repo/TextFileRecordHandler.java

```diff
@@ -107,7 +107,13 @@ public void setParams(Map<String, String> params) throws IllegalArgumentExceptio
 	 * @return null if no sanitization needed, else the new id
 	 */
 	private String sanitizeID(String id) {
-		String s = id.replaceAll("\\n", "_-_NEWLINE_-_").replaceAll("\\r", "_-_RETURN_-_").replaceAll("\\t", "_-_TAB_-_").replaceAll(" ", "_-_SPACE_-_").replaceAll("\\\\", "_-_BACKSLASH_-_").replaceAll("/", "_-_FORWARDSLASH_-_").replaceAll(":", "_-_COLON_-_").replaceAll("\\*", "_-_STAR_-_").replaceAll("\\?", "_-_QUESTIONMARK_-_").replaceAll("\"", "_-_DOUBLEQUOTE_-_").replaceAll("<", "_-_LESSTHAN_-_").replaceAll(">", "_-_GREATERTHAN_-_").replaceAll("\\|", "_-_PIPE_-_");
+		String urlDecodedId = "";
+		try {
+			urlDecodedId = java.net.URLDecoder.decode(id, java.nio.charset.StandardCharsets.UTF_8.name());
+		} catch (Exception e) {
+			// not going to happen - value came from JDK's own StandardCharsets
+		}
+		String s = urlDecodedId.replaceAll("\\n", "_-_NEWLINE_-_").replaceAll("\\r", "_-_RETURN_-_").replaceAll("\\t", "_-_TAB_-_").replaceAll(" ", "_-_SPACE_-_").replaceAll("\\\\", "_-_BACKSLASH_-_").replaceAll("/", "_-_FORWARDSLASH_-_").replaceAll(":", "_-_COLON_-_").replaceAll("\\*", "_-_STAR_-_").replaceAll("\\?", "_-_QUESTIONMARK_-_").replaceAll("\"", "_-_DOUBLEQUOTE_-_").replaceAll("<", "_-_LESSTHAN_-_").replaceAll(">", "_-_GREATERTHAN_-_").replaceAll("\\|", "_-_PIPE_-_");
```

Comment:
I suggest to sanitize also . (dot) in the identifier. Some file systems might have an issue with dot in the name of the file.

Reply:
I added the `.` escape